moveit_commander/src/moveit_commander/move_group.py

```diff
@@ -761,18 +720,16 @@ def retime_trajectory(
         acceleration_scaling_factor=1.0,
         algorithm="iterative_time_parameterization",
     ):
-        ser_ref_state_in = conversions.msg_to_string(ref_state_in)
-        ser_traj_in = conversions.msg_to_string(traj_in)
-        ser_traj_out = self._g.retime_trajectory(
-            ser_ref_state_in,
-            ser_traj_in,
-            velocity_scaling_factor,
-            acceleration_scaling_factor,
-            algorithm,
-        )
-        traj_out = RobotTrajectory()
-        traj_out.deserialize(ser_traj_out)
-        return traj_out
+        try:
+            return self._g.retime_trajectory(
+                ref_state_in,
+                traj_in,
+                velocity_scaling_factor,
+                acceleration_scaling_factor,
+                algorithm,
+            )
+        except CppDeserializationError as e:
+            raise DeserializationError(e)
```

Comment:
The DeserializationError could be observed when a empty ByteString was returned from the wrapper.
There is actually a test in moveit_commander.
I changed the wrapper to directly throw this exception, so that the behavior stays the same